src/link/object.c

```diff
@@ -459,7 +459,12 @@ static struct Section *getMainSection(struct Section *section)
 
 void obj_ReadFile(char const *fileName, unsigned int fileID)
 {
-	FILE *file = strcmp("-", fileName) ? fopen(fileName, "rb") : stdin;
+	FILE *file;
+
+	if (strcmp("-", fileName) != 0)
+		file = fopen(fileName, "rb");
+	else
+		file = fdopen(STDIN_FILENO, "rb");
```

Comment:
I'm surprised that this is available on Windows.
```suggestion
		file = fdopen(STDIN_FILENO, "rb"); // `stdin` is in text mode by default
```

Reply:
They're in include/platform.h.

Reply:
`fdopen`, not `*_FILENO`.

Reply:
Ah. It's apparently [a deprecated alias](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/fdopen?view=msvc-170) for `_fdopen`. We might want to add `# define fdopen _fdopen` in platform.h the way we already do for `strdup`.